src/Tests/dotnet-workload-install.Tests/GivenDotnetWorkloadInstall.cs

```diff
@@ -309,6 +309,52 @@ public void GivenWorkloadInstallItDoesNotRemoveOldInstallsOnRollback(bool userLo
+            var manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "MockWorkloadsSample.json");
+            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(new[] { manifestPath }), dotnetRoot, userProfileDir: userProfileDir);
+            var manifestUpdater = new MockWorkloadManifestUpdater();
+            var prev7SdkFeatureVersion = "6.0.100-preview.7.21379.14";
+            var prev7FormattedFeatureVersion = "6.0.100-preview.7";
+            var rc1SdkFeatureVersion = "6.0.100-rc.1.21463.6";
+            var rc1FormattedFeatureVersion = "6.0.100-rc.1";
+            var existingWorkload = "mock-1";
+
+            // Install a workload for preview 7
+            var installParseResult = Parser.Instance.Parse(new string[] { "dotnet", "workload", "install", existingWorkload });
+            var installCommand = new WorkloadInstallCommand(installParseResult, reporter: _reporter, workloadResolver: workloadResolver, nugetPackageDownloader: new MockNuGetPackageDownloader(tmpDir),
+                workloadManifestUpdater: manifestUpdater, userProfileDir: userProfileDir, version: prev7SdkFeatureVersion, dotnetDir: dotnetRoot, tempDirPath: testDirectory);
+            installCommand.Execute();
+
+            // Install workload for RC1
+            installCommand = new WorkloadInstallCommand(installParseResult, reporter: _reporter, workloadResolver: workloadResolver, nugetPackageDownloader: new MockNuGetPackageDownloader(tmpDir),
+                workloadManifestUpdater: manifestUpdater, userProfileDir: userProfileDir, version: rc1SdkFeatureVersion, dotnetDir: dotnetRoot, tempDirPath: testDirectory);
+            installCommand.Execute();
+
+            // Existing installation is present
+            var prev7InstallRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", prev7FormattedFeatureVersion, "InstalledWorkloads");
+            Directory.GetFiles(prev7InstallRecordPath).Count().Should().Be(1);
+            File.Exists(Path.Combine(prev7InstallRecordPath, existingWorkload))
+                .Should().BeTrue();
+
+            var rc1InstallRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", rc1FormattedFeatureVersion, "InstalledWorkloads");
+            Directory.GetFiles(rc1InstallRecordPath).Count().Should().Be(1);
+            File.Exists(Path.Combine(rc1InstallRecordPath, existingWorkload))
+                .Should().BeTrue();
+
+            var packRecordDirs = Directory.GetDirectories(Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", "v1"));
+            packRecordDirs.Count().Should().Be(3);
+            var featureBandRecords = Directory.GetFiles(Directory.GetDirectories(packRecordDirs[0])[0]);
+            featureBandRecords.Count().Should().Be(2);
+            var installPacks = Directory.GetDirectories(Path.Combine(dotnetRoot, "packs"));
+            installPacks.Count().Should().Be(2);
```

Comment:
What are the different packs and directories that are being checked here?  Some comments or possibly some more assertions about the actual values would be nice.